src/runtime_src/core/edge/user/shim.cpp

```diff
@@ -338,6 +350,42 @@ void *ZYNQShim::xclMapBO(unsigned int boHandle, bool write)
   return ptr;
 }
 
+void *ZYNQShim::xclAllocHostPtr(size_t size, unsigned flags)
+{
+  std::cout << "ZYNQShim::xclAllocHostPtr called" << std::endl;
+  unsigned int boHandle = xclAllocBO(size, 0, flags);
+  void* vaddr =  xclMapBO(boHandle, true);
+  //printf("%s:%d vaddr:%x\n",__func__,__LINE__,(uint64_t)vaddr);
+  xclBOProperties properties;
+  int ret = xclGetBOProperties(boHandle, &properties);
+  if(vaddr) {
+    BOData data;
+    data.boHandle = boHandle;
+    data.paddr = properties.paddr;
+    data.size = properties.size;
+    std::lock_guard<std::mutex> l(mBOMapLock);
+    auto it = mBoMap.find(reinterpret_cast<uint64_t>(vaddr));
+    if (it == mBoMap.end()) {
+      mBoMap.insert(it, std::pair<uint64_t, BOData>(reinterpret_cast<uint64_t>(vaddr), data));
+    }
+  }
+  return vaddr;
+}
+
+void ZYNQShim::xclFreeHostPtr(void* ptr)
+{
+  std::cout << "ZYNQShim::xclFreeHostPtr called" << std::endl;
+  std::lock_guard<std::mutex> l(mBOMapLock);
+  auto it = mBoMap.find(reinterpret_cast<uint64_t>(ptr));
+  if (it == mBoMap.end()) {
+    std::cout  << "xclFreeHostPtr Bad Host PTR" << std::endl;
+    return;
+  }
+  munmap(ptr,it->second.size);  
+  xclFreeBO(it->second.boHandle);
```

Comment:
Wouldn't it be better to have BOData constructor and destructor manage xclAllocBO and xclFreeBO